### PR TITLE
Re-write of Microsoft.Insights/schedulequeryrules

### PR DESCRIPTION
**Pull request by FelixBorst**
# Change

Primary goal was to update the api-version from 2018-04-16 to a newer one.
But since the newer api-versions change the complete structure of the resource, a rewrite was necessary.

Successful sample deployment: https://github.com/Azure/ResourceModules/actions/runs/1482016915

For comparison, the old api documentation...
https://docs.microsoft.com/en-us/azure/templates/microsoft.insights/2018-04-16/scheduledqueryrules?tabs=bicep

... and the new one:
https://docs.microsoft.com/en-us/azure/templates/microsoft.insights/2021-02-01-preview/scheduledqueryrules?tabs=bicep

FYI: I have used api version 2021-02-01-preview and not 2021-08-01 as the latter one is not yet supported by the bicep linter in vscode.

## Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update (Wiki)

## Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep [hunk 1]
```diff
@@ -30,148 +45,56 @@ param workspaceResourceId string
...
-}
-
 module pid_cuaId '.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
   name: 'pid-${cuaId}'
   params: {}
 }
 
-resource queryAlert 'microsoft.insights/scheduledQueryRules@2018-04-16' = {
+resource queryAlert 'Microsoft.Insights/scheduledQueryRules@2021-02-01-preview' = {
   name: alertName
```
Comment (MrMCake) [inline arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep:75]: change the `alertName` to `name`

File: arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep [hunk 2]
```diff
@@ -30,148 +45,56 @@ param workspaceResourceId string
...
-      throttlingInMin: suppressForMinutes
-      trigger: {
-        thresholdOperator: metricResultCountThresholdOperator
-        threshold: metricResultCountThreshold
-        metricTrigger: (empty(metricColumn) ? null : metricTrigger)
-      }
-      criteria: criterias
-      'odata.type': 'Microsoft.WindowsAzure.Management.Monitoring.Alerts.Models.Microsoft.AppInsights.Nexus.DataContracts.Resources.ScheduledQueryRules.${odataType}'
-    }
+    evaluationFrequency: contains(kind, 'LogAlert') ? evaluationFrequency : null
```
Comment (MrMCake) [inline arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep:90]: I'd like to understand why this is implemented like this. 

1. Why using this logic and not just reference `evaluationFrequency`?
2. Why contains?

File: arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep [hunk 3]
```diff
@@ -30,148 +45,56 @@ param workspaceResourceId string
...
+    muteActionsDuration: contains(kind, 'LogAlert') ? suppressForMinutes : null
+    overrideQueryTimeRange: contains(kind, 'LogAlert') ? queryTimeRange : null
+    scopes: scopes
+    severity: contains(kind, 'LogAlert') ? severity : null
+    skipQueryValidation: contains(kind, 'LogAlert') ? skipQueryValidation : null
+    targetResourceTypes: contains(kind, 'LogAlert') ? targetResourceTypes : null
+    windowSize: contains(kind, 'LogAlert') ? windowSize : null
   }
 }
```
Comment (MrMCake) [inline arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep:99]: Please add `@description('...')` to the outputs and regenerate the readme

File: arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep (hunk 2 above)
Comment (FelixBorst) [reply arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep:90]: The docs state, that only when the alert is of type LogAlert this parameter should be used.

File: arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep (hunk 3 above)
Comment (FelixBorst) [reply arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep:99]: Done :)

File: arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep (hunk 1 above)
Comment (FelixBorst) [reply arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep]: done :)

File: arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep (hunk 2 above)
Comment (MrMCake) [reply arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep:90]: alright. However, I guess we should change it to 
```bicep
evaluationFrequency: (kind == 'LogAlert') ? evaluationFrequency : null
```

Comment (FelixBorst) [reply arm/Microsoft.Insights/scheduledQueryRules/deploy.bicep]: I like it and I changed it ;)

**Review by MrMCake — APPROVED** [review]

**Pull request merged**